src/components/orchestrator/collector/index.js

```diff
@@ -81,13 +83,14 @@ export const Orchestrator = ({
       date: new Date().getTime(),
       currentPage: currentPage,
     };
-    return newStateData;
+    setCurrentStateData(newStateData);
   };
 
   const logoutAndClose = async () => {
     if (!validated) {
+      updateStateData();
       const dataToSave = {
-        stateData: updateStateData(),
+        stateData: currentStateData,
```

Comment:
Keep the result of function updateStateData()

Why, because of pb of synchronisation between state and execution of a function